kafka-streams-partitioners/weighted-group-partitioner/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/RoundRobinPartitioner.java

```diff
@@ -0,0 +1,13 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+
+public class RoundRobinPartitioner<K, V> implements StreamPartitioner<K, V> {
+  private final AtomicInteger currentPartition = new AtomicInteger();
+
+  @Override
+  public Integer partition(String topic, K key, V value, int numPartitions) {
+    return currentPartition.getAndSet((currentPartition.get() + 1) % numPartitions);
```

Comment:
getAndIncrement is equivalent. 

This will eventually overflow however, and go negative. The way this is written, that would start returning negative partition values right?

Reply:
>getAndIncrement is equivalent.
Nope. I don't think this will overflow. Modulo operation is performed before setting the new value.
This is be like cyclic array will be returning the value only in the range (0, 1, ..., n-1)

Reply:
Ah I misread the parentheses - confusing way to write it though. Thoughts on nit rewriting it something like 
```java
return currentPartition.getAndUpdate(previousPartition -> (previousPartition + 1) % numPartitions);
```